http-api/src/lib.rs

```diff
@@ -450,8 +466,20 @@ async fn history_handler(
 
     let commits = browse(reference, ctx.paths, |browser| {
         let mut result = radicle_source::commits::<PeerId>(browser, None)?;
-        let headers: Vec<Header> = result
+
+        // To create chunks that make sense we sort the commits from newest to oldest.
+        result
             .headers
+            .sort_by(|a, b| b.committer_time.cmp(&a.committer_time));
+        let paginated_headers: Vec<Vec<Header>> = result
+            .headers
+            .chunks(per_page.unwrap_or_default())
+            .map(|x| x.to_vec())
+            .collect();
```

Comment:
We don't want to actually create all pages if we're only going to get the first page for eg. (imagine a large commit history with 100K+ commits). So instead of using `collect()` and then `get()` we should be able to use `.nth()` on the iterator.

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.nth

Reply:
Yeah you're right, will fix this